View/List

snowflake-1117 opened the pull request:
뷰 폴더에 있는 Animation 클래스를 코틀린으로 재작성 해보았습니다.

안드로이드도 공부하는 중이고 코틀린도 공부하는 중이라 어떤 부분을 어떻게 손대야 할지 막막했지만 나름대로 고쳐봤어요!

snowflake-1117 committed:
Add component to .gitignore file

snowflake-1117 committed:
Convert Animation1 Java file into Kotiln file

snowflake-1117 committed:
Use android:inputType instead of android:password

Because android:password in EditText is Deprecated.

snowflake-1117 committed:
Convert Animation2 Java file into Kotlin file

snowflake-1117 committed:
Convert Animation3 Java file into Kotiln file

snowflake-1117 committed:
Remove unnecessary variable declaration using let

snowflake-1117 committed:
Move variables declaration location for readability

snowflake-1117 committed:
Remove unnecessary variable declaration using let

snowflake-1117 committed:
Initialize variable a using apply

snowflake-1117 committed:
Variable names were replaced by meaningful names

Pluu commented:
👍

snowflake-1117 committed:
extensions 플러그인 추가

snowflake-1117 committed:
extensions 적용

- findViewById 제거
- run 사용
- ViewFlipper 변수 lateinit 적용: nullable -> notNull

snowflake-1117 committed:
AutoComplete1~3 코틀린으로 변환

snowflake-1117 commented:
Kotlin android extensions 적용

snowflake-1117 committed:
불필요한  run 적용 제거

Pluu commented on mobile/src/main/java/com/example/android/apis/view/Animation2.kt:
```diff
@@ -0,0 +1,85 @@
+        ArrayAdapter(this,
+                android.R.layout.simple_spinner_item, mStrings).let {
+            it.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
+            spinner.run { adapter = it }
```
spinner.run을 사용한다면 아래 onItemSelectedListener  도 한번에 묶는게 좋을듯합니다. scope 변환로직이 매번 일어날테니

snowflake-1117 replied:
앗 네! 오늘 보니까 그렇더라구요 ㅋㅋㅋㅋ 감사합니다

Pluu commented:
```
val (inAnim, outAnim) = when (position) {
            0 -> {
                arrayOf(AnimationUtils.loadAnimation(this, R.anim.push_up_in),
                        AnimationUtils.loadAnimation(this, R.anim.push_up_out))
            }
            1 -> {
                arrayOf(AnimationUtils.loadAnimation(this, R.anim.push_left_in),
                        AnimationUtils.loadAnimation(this, R.anim.push_left_out))
            }
            2 -> {
                arrayOf(AnimationUtils.loadAnimation(this, android.R.anim.fade_in),
                        AnimationUtils.loadAnimation(this, android.R.anim.fade_out))
            }
            else -> {
                arrayOf(AnimationUtils.loadAnimation(this, R.anim.hyperspace_in),
                        AnimationUtils.loadAnimation(this, R.anim.hyperspace_out))
            }
        }
        mFlipper!!.inAnimation = inAnim
        mFlipper!!.outAnimation= outAnim
```

when 내부에 Pair 처리가 되면 편할텐데, 잘안되서 arrayOf로 한번 바꿔봤습니다

snowflake-1117 commented:
when 내부를 뭔가 바꿔보고 싶은 생각이 자꾸 들었는데 어떻게 해야 할지 모르겠더라구요. 그런데 이런 방법이 있었군요! 벌써 두번씩이나 ㅋㅋ 리뷰 감사합니다!!

snowflake-1117 committed:
when 내부 pair 처리를 위해 arrayOf 사용

(Pluu 님의 리뷰)

snowflake-1117 committed:
animation.interpolator에 when의 결과 값을 할당해 코드를 간결하게 함

snowflake-1117 committed:
alpha9 업데이트 적용

snowflake-1117 committed:
AutoComplete6 코틀린으로 변환

snowflake-1117 committed:
AutoComplete4 코틀린으로 변환

- READ_CONTECTS permission denied

snowflake-1117 committed:
READ_CONTACTS 권한 permission 기능 구현

snowflake-1117 committed:
자동완성이 되지 않는 버그 수정

입력받는 문자열이 빈칸일 때 W/Filter: An exception occured during performFiltering()! java.lang.IllegalArgumentException: Parameter specified as non-null is null 문제 발생

snowflake-1117 committed:
권한 허용 여부를 체크하는 checkPermission 함수 분리

snowflake-1117 committed:
코드 리팩토링

snowflake-1117 committed:
권한허용 함수 적용

EditText에 '*'를 입력 해도 전체 Contacts 보이지 않음

snowflake-1117 committed:
Baseline 파일들 코틀린으로 변환

snowflake-1117 committed:
BaselineNaested 파일들 코틀린으로 변환

snowflake-1117 committed:
자바파일 코틀린 파일로 변환

snowflake-1117 committed:
Revert "자바파일 코틀린 파일로 변환"

This reverts commit 4f2fe65eda2c10f6d779710ffbe971f3b254886d.

snowflake-1117 committed:
gradle 업데이트 적용

snowflake-1117 committed:
매니페스트 파일에 맞춰 minSdkVersion 23으로 변경

snowflake-1117 committed:
다이얼로그 출력 문자열 리소스 파일로 이동 후 적용

Pluu commented on build.gradle:
```diff
@@ -7,7 +7,7 @@ buildscript {
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.0.0-alpha9'
+        classpath 'com.android.tools.build:gradle:3.0.0-beta2'
```
👍 빠른 도입

Pluu commented on mobile/build.gradle:
```diff
@@ -4,11 +4,11 @@ apply plugin: 'kotlin-android-extensions'
 android {
     compileSdkVersion 26
-    buildToolsVersion "26.0.0"
+    buildToolsVersion '26.0.0'
```
26.0.0의 버그가 수정된 `26.0.1` 이 나온 상태

Pluu commented on mobile/src/main/java/com/example/android/apis/view/AutoComplete4.kt:
```diff
@@ -16,43 +16,121 @@
-        val cursor = content.query(Contacts.CONTENT_URI,
-                CONTACT_PROJECTION, null, null, null)
+        // Here, thisActivity is the current activity
+        if (ContextCompat.checkSelfPermission(this,
```
퍼미션 체크 로직은 자주 사용되므로

- https://github.com/googlesamples/easypermissions
- http://blog.dramancompany.com/2015/11/리멤버의-안드로이드-6-0-m버전-대
응기/

응용하는 방법도 좋을듯

snowflake-1117 committed:
입력받는 문자열이 빈칸일 때, non-null로 추정되는 변수가 null인 예외 발생문제 해결

snowflake-1117 committed:
buildToolsVersion 26.0.1 적용

snowflake-1117 committed:
연락처 Permission 체크를 위한 공통 클래스 생성

snowflake-1117 committed:
ContactPermission 클래스 적용

snowflake-1117 committed:
gradle 업데이트

snowflake-1117 committed:
List1 코틀린으로 변환

snowflake-1117 committed:
List2 코틀린으로 변환

deprecated 함수 삭제 및 로더 대체 적용

snowflake-1117 committed:
lazy 사용해서 var -> val 변수로 전환

init 사용 시 NullPointerException 에러 -> contentResolver가 context를 받지 못하기 때문이었다.

snowflake-1117 committed:
List3 코틀린으로 변환

snowflake-1117 committed:
List4 코틀린으로 변환

snowflake-1117 committed:
List5 코틀린으로 변환

inner class로 변환됨 -> 바깥 scope에 있는 mStrings를 access 하기 위함

snowflake-1117 committed:
List6 코틀린으로 변환

apply, when 사용

snowflake-1117 committed:
List7 코틀린으로 변환

snowflake-1117 committed:
List8 코틀린으로 변환

snowflake-1117 committed:
List8 getView 파라미터 convertView Nullable로 전환

snowflake-1117 committed:
List9 코틀린으로 변환

snowflake-1117 committed:
List10 코틀린으로 변환

snowflake-1117 committed:
List11 코틀린으로 변환

snowflake-1117 committed:
List12 코틀린으로 변환

snowflake-1117 committed:
List13 코틀린으로 변환

snowflake-1117 committed:
List14 코틀린으로 변환

snowflake-1117 committed:
List15 코틀린으로 변환

snowflake-1117 committed:
List16 코틀린으로 변환

snowflake-1117 committed:
사용하지 않는 import 삭제

snowflake-1117 committed:
List17 코틀린으로 변환

snowflake-1117 commented:
View 폴더에 있는 List 파일들 코틀린으로 변환했습니다.
